--- a/uboot-mtk-20230718-09eda825/cmd/glbtn.c
+++ b/uboot-mtk-20230718-09eda825/cmd/glbtn.c
@@ -48,26 +48,26 @@

 static int do_glbtn(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 {
-	const char *button_label = NULL;
-	int ret, counter = 0;
-	struct udevice *dev;
+	const char *button_label = NULL;
+	int ret, counter = 0;
+	struct udevice *dev;
 	ulong ts;

 	led_control("ledblink", "blink_led", "250");

 	gpio_power_clr();

-	button_label = env_get("glbtn_key");
-	if (!button_label)
-		button_label = "reset";
-
-	ret = button_get_by_label(button_label, &dev);
-	if (ret) {
-		printf("Button '%s' not found (err=%d)\n", button_label, ret);
+	button_label = env_get("glbtn_key");
+	if (!button_label)
+		button_label = "mesh";
+
+	ret = button_get_by_label(button_label, &dev);
+	if (ret) {
+		printf("Button '%s' not found (err=%d)\n", button_label, ret);
 		return CMD_RET_FAILURE;
 	}

-	if (!button_get_state(dev)) {
+	if (!button_get_state(dev)) {
 		poller_async_register(&led_p, "led_pa");
 		poller_call_async(&led_p, 1000000, led_action_post, NULL);
 		return CMD_RET_SUCCESS;
===================================================================
--- a/atf-20240117-bacca82a8/plat/mediatek/mt7981/drivers/pll/pll.c
+++ b/atf-20240117-bacca82a8/plat/mediatek/mt7981/drivers/pll/pll.c
@@ -94,11 +94,20 @@ void mtk_pll_init(int skip_dcm_setting)
 	mmio_clrbits_32(MPLL_PWR_CON0, CON0_ISO_EN);
 
 	/* Set PLL frequency */
-	mmio_write_32(ARMPLL_CON1, 0x82000000); /* 1.3G */
-
-	//mmio_setbits_32(ARMPLL_CON0, 0x124); /* divider for 650M */
-
-	mmio_setbits_32(ARMPLL_CON0, 0x114); /* divider for 1.3G */
+	// 251218: overclock hack
+	NOTICE("THIS IS AN OVERCLOKING BL2!\n");
+	NOTICE("set arm freq to 1.60 GHz:\n");
+
+	// mmio_write_32(ARMPLL_CON1, 0x82000000); /* 1.3G */
+	mmio_write_32(ARMPLL_CON1, 0x50000000); /* 1.6G */
+	// mmio_write_32(ARMPLL_CON1, 0x55000000); /* 1.7G */
+	NOTICE("ARMPLL_CON1 = 0x%x\n", mmio_read_32(ARMPLL_CON1));
+
+	// mmio_setbits_32(ARMPLL_CON0, 0x124); /* divider for 650M */
+	// mmio_setbits_32(ARMPLL_CON0, 0x114); /* divider for 1.3G */
+	mmio_clrbits_32(ARMPLL_CON0, 0x70); /* divider for >1.3G */
+	mmio_setbits_32(ARMPLL_CON0, 0x104);
+	NOTICE("ARMPLL_CON0 = 0x%x\n", mmio_read_32(ARMPLL_CON0));
 
 	mmio_setbits_32(NET2PLL_CON0, 0x114);
 	mmio_setbits_32(MMPLL_CON0, 0x124);
===================================================================
--- a/uboot-mtk-20230718-09eda825/arch/arm/dts/mt7981-ax3000t.dts
+++ b/uboot-mtk-20230718-09eda825/arch/arm/dts/mt7981-ax3000t.dts
@@ -40,13 +40,6 @@
 			mtdids = "nmbm0=nmbm0";
 			mtdparts = "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),112m(ubi),256k(KF)";
 		};
-
-		layout@2 {
-			label = "qwrt";
-			// mtd layout for lean qwrt
-			mtdids = "nmbm0=nmbm0";
-			mtdparts = "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),100m(ubi),12m(data),256k(KF)";
-		};
 	};

 	gpio-keys {
