--- a/uboot-mtk-20250711/arch/arm/dts/mt7981-xiaomi-ax3000t.dts
+++ b/uboot-mtk-20250711/arch/arm/dts/mt7981-xiaomi-ax3000t.dts
@@ -40,13 +40,6 @@
 			mtdids = "nmbm0=nmbm0";
 			mtdparts = "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),112m(ubi),256k(KF)";
 		};
-
-		layout@2 {
-			label = "qwrt";
-			// mtd layout for lean qwrt
-			mtdids = "nmbm0=nmbm0";
-			mtdparts = "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),100m(ubi),12m(data),256k(KF)";
-		};
 	};

 	gpio-keys {
