--- a/uboot-mtk-20250711/arch/arm/dts/mt7981-xiaomi-ax3000t.dts
+++ b/uboot-mtk-20250711/arch/arm/dts/mt7981-xiaomi-ax3000t.dts
@@ -40,13 +40,6 @@
 			mtdids = "nmbm0=nmbm0";
 			mtdparts = "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),112m(ubi),256k(KF)";
 		};
-
-		layout@2 {
-			label = "qwrt";
-			// mtd layout for lean qwrt
-			mtdids = "nmbm0=nmbm0";
-			mtdparts = "nmbm0:1024k(bl2),256k(Nvram),256k(Bdata),2048k(factory),2048k(fip),256k(crash),256k(crash_log),100m(ubi),12m(data),256k(KF)";
-		};
 	};

 	gpio-keys {
===================================================================
--- a/atf-20240117-bacca82a8/plat/mediatek/mt7981/drivers/pll/pll.c
+++ b/atf-20240117-bacca82a8/plat/mediatek/mt7981/drivers/pll/pll.c
@@ -19,7 +19,7 @@
 #define BUS_PLL_DIVIDER 		0x104007c0

 #ifndef MT7981_ARMPLL_FREQ_MHZ
-#define MT7981_ARMPLL_FREQ_MHZ		1300U
+#define MT7981_ARMPLL_FREQ_MHZ		1600U
 #endif

 #define ARMPLL_FREQ_STEP_MHZ		20U
