--- a/cmd/glbtn.c
+++ b/cmd/glbtn.c
@@ -22,7 +22,8 @@ static int do_glbtn(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 {
-	const char *button_label = NULL;
 	int ret, counter = 0;
-	struct udevice *dev;
+	struct udevice *dev_reset = NULL;
+	struct udevice *dev_mesh = NULL;
 	ulong ts;

 	led_control("ledblink", "blink_led", "250");

@@ -32,17 +33,22 @@ static int do_glbtn(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 
-	button_label = env_get("glbtn_key");
-	if (!button_label)
-		button_label = "reset";
-
-	ret = button_get_by_label(button_label, &dev);
-	if (ret) {
-		printf("Button '%s' not found (err=%d)\n", button_label, ret);
-		return CMD_RET_FAILURE;
-	}
+	ret = button_get_by_label("reset", &dev_reset);
+	if (ret)
+		dev_reset = NULL;
+
+	ret = button_get_by_label("mesh", &dev_mesh);
+	if (ret)
+		dev_mesh = NULL;
+
+	if (!dev_reset && !dev_mesh) {
+		printf("Buttons 'reset' and 'mesh' not found\n");
+		return CMD_RET_FAILURE;
+	}

-	if (!button_get_state(dev)) {
+	if ((!dev_reset || !button_get_state(dev_reset)) &&
+	    (!dev_mesh || !button_get_state(dev_mesh))) {
 		poller_async_register(&led_p, "led_pa");
 		poller_call_async(&led_p, 1000000, led_action_post, NULL);
 		return CMD_RET_SUCCESS;
@@ -52,11 +58,12 @@ static int do_glbtn(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 
-	printf("RESET button is pressed for: %2d second(s)", counter++);
+	printf("BUTTON is pressed for: %2d second(s)", counter++);

 	ts = get_timer(0);

-	while (button_get_state(dev) && counter < 4) {
+	while (((dev_reset && button_get_state(dev_reset)) ||
+		(dev_mesh && button_get_state(dev_mesh))) &&
+	       counter < 4) {
 		if (get_timer(ts) < 1000)
 			continue;
