--- a/uboot-mtk-20230718-09eda825/cmd/glbtn.c
+++ b/uboot-mtk-20230718-09eda825/cmd/glbtn.c
@@ -48,26 +48,26 @@

 static int do_glbtn(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 {
-	const char *button_label = NULL;
-	int ret, counter = 0;
-	struct udevice *dev;
+	const char *button_label = NULL;
+	int ret, counter = 0;
+	struct udevice *dev;
 	ulong ts;

 	led_control("ledblink", "blink_led", "250");

 	gpio_power_clr();

-	button_label = env_get("glbtn_key");
-	if (!button_label)
-		button_label = "reset";
-
-	ret = button_get_by_label(button_label, &dev);
-	if (ret) {
-		printf("Button '%s' not found (err=%d)\n", button_label, ret);
+	button_label = env_get("glbtn_key");
+	if (!button_label)
+		button_label = "mesh";
+
+	ret = button_get_by_label(button_label, &dev);
+	if (ret) {
+		printf("Button '%s' not found (err=%d)\n", button_label, ret);
 		return CMD_RET_FAILURE;
 	}

-	if (!button_get_state(dev)) {
+	if (!button_get_state(dev)) {
 		poller_async_register(&led_p, "led_pa");
 		poller_call_async(&led_p, 1000000, led_action_post, NULL);
 		return CMD_RET_SUCCESS;
